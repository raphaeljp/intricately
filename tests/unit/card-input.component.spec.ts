import { shallowMount } from '@vue/test-utils';
import CardInput from '@/components/card-input.component.vue';

const factory = (values = {}, props = {}) => shallowMount(CardInput, {
  data() { return { ...values }; },
  propsData: { id: '1', ...props },
});

describe('Card Input Component', () => {
  it('should be mounted', () => {
    const wrapper = factory();
    expect(wrapper).toBeTruthy();
  });

  it('should not have a label if props "hideLabel" is passed', () => {
    const wrapper = factory({}, { hideLabel: true });
    const label = wrapper.find('label');
    expect(label.exists()).toBe(false);
  });

  test('renders correctly', () => {
    const wrapper = factory();
    expect(wrapper).toMatchSnapshot();
  });
});
