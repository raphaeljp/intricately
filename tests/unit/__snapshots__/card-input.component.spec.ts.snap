// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Input Component renders correctly 1`] = `
<div class="card-input__box"><label for="card-input-1">CARD 11</label> <input id="card-input-1" type="text" maxlength="3">
  <!---->
</div>
`;
